### PR TITLE
App bar

### PR DESCRIPTION
**Pull request by aninternetof**
This replaces the top navigation bar with a one from Material UI. I also tried to make it look like the mocks.

### CODE REVIEW

File: src/components/TopNav.js [hunk 1]
```diff
@@ -3,17 +3,21 @@ import { connect } from 'react-redux';
...
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import Grid from '@material-ui/core/Grid';
+import { Box } from '@material-ui/core';
 import { FormattedMessage } from 'react-intl';
 import { withCookies, Cookies } from 'react-cookie';
 import { logout as actionLogout } from '@/actions/auth';
 
-import logoImage from '@/assets/images/rovercode_logo_magenta.png';
```
Comment (aninternetof) [inline src/components/TopNav.js:15]: I don't have the logo/connection area done from the mocks. That's coming in the next PR.

File: src/components/TopNav.js [hunk 2]
```diff
@@ -36,9 +41,27 @@ class TopNav extends Component {
...
+      userMenuAnchorElement: null,
+    });
+  };
+
   render() {
     const { userName } = this.props;
-    const { redirectToLogin } = this.state;
+    const { redirectToLogin, userMenuAnchorElement } = this.state;
+
+    const NavBarSpacer = withStyles((theme) => ({
```
Comment (aninternetof) [inline src/components/TopNav.js:60]: Adding this margin directly to the `AppBar` somehow broke the user drop-down menu. I think this is probably about as good.

File: src/components/TopNav.js [hunk 3]
```diff
@@ -47,39 +70,150 @@ class TopNav extends Component {
...
-            <Dropdown item text={userName}>
-              <Dropdown.Menu>
-                <Dropdown.Item as={Link} to="/user/settings">
+
+        <div>
+          <AppBar position="static" color="secondary">
+            <Toolbar>
+              <Grid item container direction="row" spacing={4}>
+                <Grid item xs={2}>
+                  <Box
```
Comment (aninternetof) [inline src/components/TopNav.js:79]: The mock called for the nav links to be centered inside areas with even widths. I did that by setting the grid item at `{2}` columns and using a `Box` inside.
The `{2}`-column width looks pretty good all the down to tablet sizes, so I kept it constant.

**Comment by aninternetof** [comment]
Semantic is making the link text turn blue on hover. That should go away when we remove Semantic.

**Comment by aninternetof** [comment]
Material-UI's dropdown is technically a type of pop-up. Seems to work fine. The only slight annoyance in Chrome is that it makes the page's vertical scrollbar disappear, which causes a slight screen resize. Maybe we could fix that somehow sometime.

File: src/components/TopNav.js [hunk 4]
```diff
@@ -3,17 +3,21 @@ import { connect } from 'react-redux';
...
+} from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
+import Menu from '@material-ui/core/Menu';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import Grid from '@material-ui/core/Grid';
+import { Box } from '@material-ui/core';
```
Comment (cabarnes) [inline src/components/TopNav.js:16]: Can all of the non-icon imports above be grouped with this one?

Comment (aninternetof) [reply src/components/TopNav.js:16]: Yup! I'll fix that. I'm surprised that's not something that eslint can recognize and flag.

Comment (cabarnes) [reply src/components/TopNav.js:16]: I prefer one per line like before so it's easier to see changes to single ones

**Pull request merged**